VinylControlControl: Use mixxx::audio::FramePos internally

Holzhaus opened the pull request.

Holzhaus committed:
VinylControlControl: Use mixxx::audio::FramePos internally

uklotzde commented on src/engine/controls/vinylcontrolcontrol.cpp:
```diff
@@ -89,25 +89,34 @@ void VinylControlControl::slotControlVinylSeek(double fractionalPos) {
+    if (!trackEndPosition.isValid()) {
+        return;
+    }
+    const auto newPlayPos = trackEndPosition * fractionalPos;
```
`newPlayPos = kStartFramePos + (trackEndPosition - kStartFramePos) * fractionalPos;`?

I suppose there are much more spots where positions and offsets/distances are used confused. Those could only be detected and fixed by introducing a distinct type for `FrameDiff_t`.

Holzhaus replied:
5dc279b9c872ad61e74042cc9b8617e4103a431d

Holzhaus committed:
VinylControlControl: Use semantically correct fractionalPos calculation

As requested here: https://github.com/mixxxdj/mixxx/pull/4079#discussion_r666279561

uklotzde reviewed (approved):
LGTM
